tests/test_downloader_handlers.py

```diff
@@ -796,10 +799,13 @@ def test_download_with_proxy_https_noconnect(self):
         def _test(response):
             self.assertEqual(response.status, 200)
             self.assertEqual(response.url, request.url)
-            self.assertEqual(response.body, b'https://example.com')
+            self.assertTrue(
+                response.body == b'http://example.com'  # HTTP/1.x
+                or response.body == b'/'  # HTTP/2
+            )
 
         http_proxy = '%s?noconnect' % self.getURL('')
-        request = Request('https://example.com', meta={'proxy': http_proxy})
+        request = Request('http://example.com', meta={'proxy': http_proxy})
```

Comment:
Could you elaborate on the reason behind this change?

Reply:
Earlier while setting up tests with HTTP/2 I was having trouble with the *https://example.com* url. I did revert the change back to `https` in c67d6dea318d6f0915ae86d46ce367b6c1e8ee51 as it works now.